tests/integration_tests/test_hyperopt.py

```diff
@@ -301,21 +338,122 @@ def test_hyperopt_run_hyperopt(csv_filename, search_space, tmpdir, ray_cluster):
     # add hyperopt parameter space to the config
     config["hyperopt"] = hyperopt_configs
 
-    hyperopt_results = hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
-    if search_space == "random":
-        assert hyperopt_results.experiment_analysis.results_df.shape[0] == RANDOM_SEARCH_SIZE
-    else:
-        # compute size of search space for grid search
-        grid_search_size = 1
-        for k, v in search_parameters.items():
-            grid_search_size *= len(v["values"])
-        assert hyperopt_results.experiment_analysis.results_df.shape[0] == grid_search_size
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
@justinxzhao I had done that initially, but in a prior code review for this PR, there was a suggestion to use the `tempfile` package instead of the tmpdir fixture. Any strong reasons to use the fixture instead of `tempfile`?

I think the biggest advantage of `tempfile` is that the cleanup code is no longer required which is nice

Reply:
I believe the `tmpdir` pytext fixture also takes care of deleting any files that were written under `tmpdir`. Perhaps the old code manually deleting files in the `tmpdir` was redundant.

Reply:
@justinxzhao Just checked and you're right, the old code was redundant since the tmpdir  pytext fixture uses tempfile internally :) Great catch!